tests/api/utils/singletons/test_k8s_utils.py

```diff
@@ -58,8 +68,68 @@ def test_get_logger_pods_label_selector(
     if extra_selector:
         selector += f",{extra_selector}"
 
-    k8s_helper = server.api.utils.singletons.k8s.K8sHelper(namespace, silent=True)
     k8s_helper.list_pods = unittest.mock.MagicMock()
 
     k8s_helper.get_logger_pods(project, uid, run_type)
-    k8s_helper.list_pods.assert_called_once_with(namespace, selector=selector)
+    k8s_helper.list_pods.assert_called_once_with(
+        k8s_helper.namespace, selector=selector
+    )
+
+
+@pytest.mark.parametrize(
+    "k8s_secret_data,secrets_data",
+    [
+        ({"key1": "value1", "key2": "value2"}, []),
+        ({"key1": "value1", "key2": "value2"}, None),
+        (None, ["key1"]),
+        ({}, ["key1"]),
+    ],
+)
+def test_delete_secrets_no_changes(k8s_helper, k8s_secret_data, secrets_data):
```

Comment:
```suggestion
def test_no_secret_changes_when_keys_do_not_overlap(k8s_helper, k8s_secret_data, secrets_data):
```

Reply:
@yaelgen I prefer when test names start with the function being tested, like `test_delete_secrets`
This makes it easier to find tests related to the deletion functionality. How about naming this test `test_delete_secrets_no_secret_changes_when_keys_do_not_overlap`?

Reply:
Actually, I'm thinking maybe this all can fit into a single combined `test_delete_secrets`  to cover all scenarios, wdyt?

If you prefer separate tests, both suggested names work for me. I would prefer: `test_delete_secrets_no_changes_with_no_key_overlap` and `test_delete_secrets_confirms_deletion_for_matching_keys`
IMO it makes it  a bit cumbersome, but np.